### ISSUE TITLE
bad results

### ISSUE THREAD

**Issue by wd117** [jsll/pytorch_6dof-graspnet#17]
hello,
Thank you very much for sharing this great work.
I have trained a VAE model following the instruction,it seems work well on the NPY dataset you given
![6AF97E47358686FB544EE3B3DAF3866E](https://user-images.githubusercontent.com/58775475/131944248-d49d9102-dca3-4ea8-8de1-da8087ec15b6.jpg)
But if I use the realsense d435i get the pointcloud data,the results are not ideal
![930449C1F06619E9E86FD21DD1A25104](https://user-images.githubusercontent.com/58775475/131944587-0d2af701-a22a-4841-af1d-1de75f7c99af.jpg)
![867CBAA4350AC6521E6A3783EB240CC2](https://user-images.githubusercontent.com/58775475/131944603-9273f646-c2b8-4f71-8d03-a88068292fb4.jpg)
Obviously i have made some mistakes,but i don't know where it happens,Where do you think i'm most likely to make mistakes,hope to get your help ,thank you very much.

**Comment by jsll** [comment]
One idea, have you normalized the point-cloud from the realsense? That is, have you subtracted the mean of the input point-cloud?

**Comment by wd117** [comment]
> One idea, have you normalized the point-cloud from the realsense? That is, have you subtracted the mean of the input point-cloud?

I find the grasps are generated by the function ' estimator.generate_and_refine_grasps() ',and the only input is the pointcloud. In the function,the first step is 'prepare pc',i think during this the pc should have been normalized.

**Comment by jsll** [comment]
You are correct. Been a while since I used this code base so I have forgotten some details.

Maybe the problem is that you are providing the network with the whole point-cloud that includes both the object to grasp and the surface it rests on? As far as I remember, you should only provide the network with the object point-cloud so the first thing you have to do is to segment out the object from the surface.

**Comment by wd117** [comment]
> You are correct. Been a while since I used this code base so I have forgotten some details.
> 
> Maybe the problem is that you are providing the network with the whole point-cloud that includes both the object to grasp and the surface it rests on? As far as I remember, you should only provide the network with the object point-cloud so the first thing you have to do is to segment out the object from the surface.
I have done this ,i remove the plane and only provide the object pc. I have always suspected that my model is not good，but it  work well on the test dataset,so i'm very confused

**Comment by wd117** [comment]
> You are correct. Been a while since I used this code base so I have forgotten some details.
> 
> Maybe the problem is that you are providing the network with the whole point-cloud that includes both the object to grasp and the surface it rests on? As far as I remember, you should only provide the network with the object point-cloud so the first thing you have to do is to segment out the object from the surface.

I have done this ,i remove the plane and only provide the object pc. I have always suspected that my model is not good，but it  work well on the test dataset,so i'm very confused
